### ISSUE TITLE
[Feature] Support QwenImageEditPlus series attention mask for NPU

### ISSUE THREAD

**Issue by zhangtao0408** [huggingface/diffusers#13016]
# Problem related
https://github.com/huggingface/diffusers/blob/7a02fadad322788c64f3b61f1bb660aef8c458a9/src/diffusers/models/transformers/transformer_qwenimage.py#L938-L946

Since PR #12702 introduced the attention mask to QwenImageEditPlus Series, the current `_native_npu` backend attention implementation does not support passing in the attention mask, which causes an error. 

https://github.com/huggingface/diffusers/blob/7a02fadad322788c64f3b61f1bb660aef8c458a9/src/diffusers/models/attention_dispatch.py#L2414-L2427

We would like to enable NPU support for QwenImageEditPlus. Based on a printed check, the mask currently contains all 1s (full attention). Is it possible to use a workaround to bypass this limitation so that Qwen-Image-Edit-Plus can run normally on the NPU?

# Solution
The `npu_fusion_attention` function supports several shapes of attention masks. We need to add a check to verify if the attention_mask is supported by the NPU backend:

1. If the mask consists entirely of 1s (indicating full attention), we can pass `None` as the attention_mask to implement full attention in FA. 
2. Otherwise, we should refer to the [official documentation](https://www.hiascend.com/document/detail/zh/Pytorch/730/apiref/torchnpuCustomsapi/docs/context/torch_npu-npu_fusion_attention.md) to add a validation check and determine whether the attention mask is valid for npu_fusion_attention.

**Comment by muziyuhui666** [comment]
I have also encountered this error on the NPU when using the latest version of diffusers (version 0.37.0.dev0).

**Comment by zhangtao0408** [comment]
> I have also encountered this error on the NPU when using the latest version of diffusers (version 0.37.0.dev0).

Cherry-pick https://github.com/huggingface/diffusers/pull/13017
